tests/epyccel/modules/python_annotations.py

```diff
@@ -0,0 +1,41 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+
+def array_int32_1d_scalar_add( x:'int32[:]', a:'int32' ):
+    x[:] += a
+
+def array_int32_2d_C_scalar_add( x:'int32[:,:]', a:'int32' ):
+    x[:,:] += a
+
+def array_int32_2d_F_add( x:'int32[:,:](order=F)', y:'int32[:,:](order=F)' ):
+    x[:,:] += y
+
+def array_int_1d_scalar_add( x:'int[:]', a:'int' ):
+    x[:] += a
+
+def array_real_1d_scalar_add( x:'real[:]', a:'real' ):
+    x[:] += a
+
+def array_real_2d_F_scalar_add( x:'real[:,:](order=F)', a:'real' ):
+    x[:,:] += a
+
+def array_real_2d_F_add( x:'real[:,:](order=F)', y:'real[:,:](order=F)'  ):
+    x[:,:] += y
+
+def array_int32_2d_F_complex_3d_expr( x:'int32[:,:](order=F)', y:'int32[:,:](order=F)' ):
+    from numpy import full, int32
+    z = full((2,3),5,order='F', dtype=int32)
+    x[:] = (x // y) * x + z
+
+def array_real_1d_complex_3d_expr( x:'real[:]', y:'real[:]' ):
+    from numpy import full
+    z = full(3,5)
+    x[:] = (x // y) * x + z
+
+def fib(n: int) -> int:
```

Comment:
Does the following work:
```
def fib(n: np.int32) -> np.int32:
```
?

Reply:
not yet, it works like the @types decorator

Reply:
Have you tried it? I think we can use `np.int32` as `dtype` for arrays so the semantics should understand it

Reply:
it's needs much more work because everything is converted to the header syntax , so we need to modify the headers in order to understand the language, or abandon the headers

Reply:
if the following works then it will automatically work here,
`#$ header function f(np.int32) results(np.int32)`
or
`@types(np.int32, np.int32)`

Reply:
what I'm doing now is convert the python type hints to the `@types` decorator and then it gets converted to a header string in the syntactic stage.
one way to solve this issue is to make the conversion in the semantics stage from the `@types` to the header language, with this we will avoid updating the header language.

Reply:
ok

Reply:
I created this issue  #633 to solve this problem